ui-modules/app-inspector/app/components/stream/stream.directive.js

```diff
@@ -67,28 +99,170 @@ export function streamDirective() {
+         */
+        function getFormattedItemLogLevel(formattedItem) {
+            if (formattedItem.isWarning) {
+                return 'log-warning';
+            } else  if (formattedItem.isError) {
+                return 'log-error';
+            } if (formattedItem.isDebug) {
+                return 'log-debug';
+            }
+            return 'log-trace';
+        }
+
+        /**
+         * @returns {boolean} True if formatted item should be displayed, and false otherwise.
+         */
+        function isDisplayFormattedItem(formattedItem) {
+            return formattedItem.isWarning && $scope.isDisplayWarning
+                || formattedItem.isDebug && $scope.isDisplayDebug
+                || formattedItem.isError && $scope.isDisplayError
+                || formattedItem.isTrace && $scope.isDisplayTrace
+                || formattedItem.isOther && $scope.isDisplayOther;
+        }
+
+        /**
+         * Formats CLI XML output and displays it in 'filtered-stream-content' field.
+         */
+        function formatCliXmlContent() {
+
+            // Slice at index of last closing tag ending wth the new line
+            let streamTags = $scope.stream.match(/<\/(.*?)>\n/g);
+            let lastClosingTagIndex  = $scope.stream.lastIndexOf(streamTags[streamTags.length-1]);
+            let newCliXmlData = $scope.stream.slice($scope.streamProcessedUpTo, lastClosingTagIndex);
+            if (!newCliXmlData) {
+                return;
+            }
+
+            $scope.streamProcessedUpTo += newCliXmlData.length;
+
+            newCliXmlData.split(/\n/g).forEach(item => {
+                let formattedItem = {
+                    id: ($scope.filteredStream.length), // ng-repeat requires unique items, array length fits the bill
+                    text: item,
+                    isOther: false,
+                    isError: false,
+                    isDebug: false,
+                    isTrace: false,
+                    isWarning: false
+                };
+
+                if (/<s s="warning">/i.test(item)) {
+                    formattedItem.isWarning = true;
+                } else if (/<s s="debug">/i.test(item)) {
+                    formattedItem.isDebug = true;
+                } else if (/<s s="verbose">/i.test(item)) {
+                    formattedItem.isTrace = true;
+                } else if (/<s s="error">/i.test(item)) {
+                    formattedItem.isError = true;
+                } else {
+                    formattedItem.isOther = true;
+                }
+
+                // Remove CLI XML string tags for know log levels
+                if (!formattedItem.isOther) {
+                    formattedItem.text = item.replace(/<s s="(.*?)">|\t/gi, '');
+                }
+
+                // Remove CLI XML markers, newlines and replace tabs with spaces
+                formattedItem.text = formattedItem.text.replace(/<\/s>|_x000[a-z0-9]_|\n/gi, '').replace(/\t/g,' ');
```

Comment:
will this work for collapsing new lines, or if a new line is in the output?  we split on `\n` to create this loop so feels like if the input is

```
<s s="error">line 1
  line 2
</s>
```

then `line 1` and `line 2` will be other iterations of this loop and so would be classed as `other` ?

Reply:
i see, we rely on the backend formatting it to single strings for us -- never mind me